object-construction-checker/tests/socket/ACSocketTest.java

```diff
@@ -479,12 +477,19 @@ private SSLSocket configureSSLSocket(@Owning SSLSocket socket, boolean isClientS
 
 
     private void updateSocketAddresses( SelectionKey sockKey ) {
-        // error: required.method.not.called
         Socket socket = ((SocketChannel) sockKey.channel()).socket();
         SocketAddress localSocketAddress = socket.getLocalSocketAddress();
         SocketAddress remoteSocketAddress = socket.getRemoteSocketAddress();
     }
 
+    // SelectionKey doesn't implement Closeable
```

Comment:
Rather than commenting this out, can we just write the correct annotation on `SelectionKey`? I'm not familiar with the API, so I'm not sure exactly what the spec is, but we ought to try to fix it if we really should report an error here.

Reply:
I don't think we can handle `SelectionKey` yet.  It has an internal pointer to a channel that a client can access by calling `channel()`.  That channel is not new, but one registered with some `Selector` in some other code.  We don't have a way of capturing that aliasing relationship with our annotations

Reply:
The checker does issue the error correctly on this code, so I'm going to uncomment it.